pkg/operator/sync.go

```diff
@@ -417,6 +425,16 @@ func (optr *Operator) waitForDaemonsetRollout(resource *appsv1.DaemonSet) error
 	})
 }
 
+func (optr *Operator) waitForControllerConfigToBeCompleted(resource *mcfgv1.ControllerConfig) error {
+	return wait.Poll(controllerConfigCompletedInterval, controllerConfigCompletedTimeout, func() (bool, error) {
+		if err := isControllerConfigCompleted(resource, optr.ccLister.Get); err != nil {
+			glog.Errorf("controllerconfig is not completed: %v", err)
+			return false, nil
```

Comment:
we want to try to poll until timeout to prevent us exiting on interruptions / transient errors. the pool time out is 1 minute, which seems not too long.